tests/test_utils.py

```diff
@@ -397,6 +397,11 @@ def matplotlib_no_gui():
     with warnings.catch_warnings():
         warnings.filterwarnings("ignore", r"Matplotlib is currently using agg.*")
 
+        # Ignore the specific UserWarning about non-interactive FigureCanvasAgg
+        warnings.filterwarnings(
+            "ignore", r"FigureCanvasAgg is non-interactive, and thus cannot be shown"
+        )
```

Comment:
I've just filtered out the additional warning but another option would be to override `plt.show` to be a no-op  
in this context or to save the non-gui plot that is generated instead of attempting to show, then restore the `plt.show` on exit.

Reply:
This should be good enough, thanks.